dynamodb-enhanced/runtime/pom.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.quarkiverse.amazonservices</groupId>
+        <artifactId>quarkus-amazon-dynamodb-enhanced-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>quarkus-amazon-dynamodb-enhanced</artifactId>
+    <name>Quarkus - Amazon Services - DynamoDB Enhanced - Runtime</name>
+    <description>Connect to Amazon DynamoDB datastore</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-netty</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.quarkiverse.amazonservices</groupId>
+            <artifactId>quarkus-amazon-common</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>dynamodb-enhanced</artifactId>
```

Comment:
I'm surprised you don't need a dependency to the actual DynamoDB extension in addition to this?

Reply:
software.amazon.awssdk:dynamodb-enhanced has a dependency to software.amazon.awssdk:dynamodb. 

In the `DynamodbEnhancedClientProducer.java` I use Arc.container(). instance(...) to ask for an instance of DynamoDbClient or DynamoDbAsyncClient. Quarkus magic uses the DynamodbClientProducer.java which is inside quarkus-amazon-dynamodb-runtime. But this is during execution so I do not need to add it as dependecy to this module. The app which wants to use `quarkus-amazon-dynamodb-enhanced` must add `quarkus-amazon-dynamodb`.

Reply:
That's a very bad idea. You should have a dependency to `quarkus-amazon-dynamodb`. (to the deployment module in the deployment module, and to the runtime module in the runtime module)

Reply:
I can add it but I do not understand why this is necessary. In both modules of `quarkus-amazon-dynamodb-enhanced` (runtime and deployment) we never use anything from `quarkus-amazon-dynamodb`. Or is it so that the user does not need to add both dependencies?

Reply:
You expect the standard DynamodbClient to work. And for that to work fully (and make sure everything is working properly in native among other things), you need the extension.

You say yourself that `The app which wants to use quarkus-amazon-dynamodb-enhanced must add quarkus-amazon-dynamodb.`. That is exactly why you need a dependency.